rotkehlchen/exchanges/coinbase.py

```diff
@@ -684,6 +663,97 @@ def _process_normal_trade(self, event: dict[str, Any], trades: list[Trade]) -> N
 
         return None
 
+    def _process_normal_trade(self, event: dict[str, Any], timestamp: Timestamp) -> Trade | None:
+        """Turns a normal coinbase transaction into a rotki trade and returns it.
+
+        Sometimes the amounts can be negative which breaks rotki's logic which is why we use abs().
+
+        https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#transaction-types
+        If the coinbase transaction is not a trade related transaction nothing happens.
+        """
+        trade_type = TradeType.deserialize(event['type'])
+        amount = AssetAmount(abs(deserialize_asset_amount(event['amount']['amount'])))
+        tx_asset = asset_from_coinbase(event['amount']['currency'], time=timestamp)
+        native_amount = abs(deserialize_asset_amount(event['native_amount']['amount']))
+        native_asset = asset_from_coinbase(event['native_amount']['currency'], time=timestamp)
+        # rate is how much you get/give in quotecurrency if you buy/sell 1 unit of basecurrency
+        rate = Price(native_amount / amount)
+
+        fee_amount = fee_asset = None
+        if 'fee' in event:
+            fee_amount = abs(deserialize_fee(event['fee']['amount']))
+            fee_asset = asset_from_coinbase(event['fee']['currency'], time=timestamp)
+
+        return Trade(
+            timestamp=timestamp,
+            location=Location.COINBASE,
+            # in coinbase you are buying/selling tx_asset for native_asset
+            base_asset=tx_asset,
+            quote_asset=native_asset,
+            trade_type=trade_type,
+            amount=amount,
+            rate=rate,
+            fee=fee_amount,  # type: ignore  # abs() doesn't propagate Fee type
+            fee_currency=fee_asset,
+            link=str(event['id']),
+        )
+
+    def _process_advanced_trade(self, event: dict[str, Any], timestamp: Timestamp) -> Trade | None:
+        """Turns an advanced_trade_fill transaction into a rotki trade returns it.
+
+        https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#transaction-types
+        If the coinbase transaction is not a trade related transaction nothing happens.
+        """
```

Comment:
this one in addition to the previous errors can also raise indexError. Please add it to the docstring since it helps later to identify unhandled errors